File: php/src/Simple/ClosedOrder.php
<?php

namespace ImmutableDatamodel\Simple;

use ImmutableDatamodel\Shared\OrderConstraint;

readonly class ClosedOrder implements Order
{
    /**
     * @param string $orderId
     * @param OrderConstraint[] $constraints
     */
    public function __construct(
        private string $orderId,
        private array  $constraints
    ) {
    }

    public function orderId(): string
    {
        return $this->orderId;
    }

    public function constraints(): array
    {
        return $this->constraints;
    }
}

File: golang/src/simple/closed_order.go
package simple

import "example/shared"

type ClosedOrder struct {
	orderId     string
	constraints []shared.OrderConstraint
}

func (co ClosedOrder) OrderId() string {
	return co.orderId
}

func (co ClosedOrder) Constraints() []shared.OrderConstraint {
	return co.constraints
}
